frontend/src/scenes/insights/Trends.cy-spec.js

```diff
@@ -0,0 +1,353 @@
+                        value: 'http://posthog.com',
+                        operator: 'exact',
+                        type: 'event',
+                    },
+                ],
+            })
+            mount()
+
+            cy.wait('@api_insight')
+                .map(helpers.getSearchParameters)
+                .should('include', {
+                    insight: 'TRENDS',
+                    interval: 'day',
+                    display: 'ActionsLineGraph',
+                    events: JSON.stringify([
+                        {
+                            id: '$pageview',
+                            name: '$pageview',
+                            type: 'events',
+                            order: 0,
+                        },
+                    ]),
+                    properties: JSON.stringify([
+                        {
+                            key: '$browser',
+                            value: 'Chrome',
+                            operator: 'exact',
+                            type: 'event',
+                        },
+                        {
+                            key: '$current_url',
+                            value: 'http://posthog.com',
+                            operator: 'exact',
+                            type: 'event',
+                        },
+                    ]),
+                })
+
+            cy.get('[data-attr="trend-line-graph"]').should('be.visible')
+            cy.contains('http://posthog.com').should('be.visible')
+        })
+
+        it('responsive to shown as', () => {
+            helpers.setLocation('/insights', {
+                insight: 'TRENDS',
+                interval: 'day',
+                display: 'ActionsLineGraph',
+                events: [
+                    {
+                        id: '$pageview',
+                        name: '$pageview',
+                        type: 'events',
+                        order: 0,
+                    },
+                ],
+                properties: [],
+                shown_as: 'Stickiness',
+            })
+            mount()
+            cy.wait('@api_insight')
+                .map(helpers.getSearchParameters)
+                .should('include', {
+                    insight: 'TRENDS',
+                    interval: 'day',
+                    display: 'ActionsLineGraph',
+                    events: JSON.stringify([
+                        {
+                            id: '$pageview',
+                            name: '$pageview',
+                            type: 'events',
+                            order: 0,
+                        },
+                    ]),
+                    properties: '[]',
+                    shown_as: 'Stickiness',
+                })
+            cy.get('[data-attr="trend-line-graph"]').should('be.visible')
+            cy.contains('Stickiness').should('be.visible')
+        })
+
+        it('responsive to breakdown', () => {
```

Comment:
We're repeating this pattern of 'heres the url, here's the request' a lot - not very apparent how it's useful.

However we can cut down the code a lot here with something like:

```js
const mountAndCheckAPI = () => {
  helpers.setLocation('/insights, given.params)
  mount()
  
  cy.wait('@api_insight').its('request.url').should('contain', toParams(given.params))
}

describe('breakdown', () => {
  given('params', () => {
                insight: 'TRENDS',
                interval: 'day',
                display: 'ActionsLineGraph',
                events: [
                    {
                        id: '$pageview',
                        name: '$pageview',
                        type: 'events',
                        order: 0,
                    },
                ],
                properties: [],
                breakdown: '$browser',
                breakdown_type: 'event',
            })

   it('displays something', () => {
     mountAndCheckAPI()
     cy.get('[data-attr="trend-line-graph"]').should('be.visible')
     cy.contains('$browser').should('be.visible')
   })
})
```

Reply:
ohh I was misunderstanding the format of `describe` and `it`. So the setup condition for a test should always be within a describe and the `it`s are checks on an interaction with the interface given those parameters. So each of the URL tests that I have should be in its own describe because they all have separate parameter setups. Is this how to organize the tests?

Reply:
I think so - the goal should be avoid repeating yourself as much as possible and keep the body of the test as lean as possible. E.g. in 'responds to multiple entities' I see a lot of near-repetition which could be avoided with the pattern above.

This is more of a readability than correctness thing - it might make sense to merge this and for me to open a diff for what I mean rather than iterating this here?